### PR TITLE
Feat: add client option to use/not use weights provided by HSDO Server

### PR DESCRIPTION
**Pull request by QuentinBtd**
### WHY ?

If you want to let HAProxy manage weights of servers with data collected by [Agent check](https://www.haproxy.com/documentation/hapee/latest/load-balancing/health-checking/agent-health-checks/), HSDO do not set weight in HAProxy, else Agent Check and HSDO will overwrite each other

### HOW ?

Add an option in HSDO client to use or not the weights managed by HSDO. 

HSDO server will continue to provide weights into DynamoDB but the client will not used it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,8 @@ Configuration that is specific to each HSDO Client, next to HAProxy.
 
 `CLIENT_HAPROXY_BACKEND_SERVER_PORT`: Port of target servers. Default to `80`.
 
+`CLIENT_USE_WEIGHTS_PROVIDED_BY_HSDO`: Use weights provide in DynamoDB by HSDO Server. Default to `true`.
```
Comment (tanguyfalconnet) [inline README.md]:
```suggestion
`CLIENT_USE_WEIGHTS`: Use weights provided by HSDO Server. Default to `true`.
```

File: src/client/haproxy.py [hunk 2]
```diff
@@ -105,14 +106,15 @@ def __addServerInBackend(self, server, bckndName, bckndbsName):
                     self.backendServerPort
                 )
             )
-            commands.append(
-                "set server %s/%s weight %s"
-                % (
-                    bckndName,
-                    bckndbsName + str(server.backendServerID),
-                    server.weight,
+            if self.useProvidedWeights == "true":
```
Comment (tanguyfalconnet) [inline src/client/haproxy.py]:
```suggestion
            if self.useProvidedWeights:
```

File: src/client/haproxy.py [hunk 3]
```diff
@@ -23,6 +23,7 @@ def __init__(self):
         self.backendBaseName = Configuration().get("CLIENT_HAPROXY_BACKEND_BASE_NAME")
         self.fallbackBackendBaseName = Configuration().get("CLIENT_HAPROXY_FALLBACK_BACKEND_BASE_NAME")
         self.ASG = Configuration().get("CLIENT_ASG_NAMES").split(",")
+        self.useProvidedWeights = Configuration().get("CLIENT_USE_WEIGHTS_PROVIDED_BY_HSDO")
```
Comment (tanguyfalconnet) [inline src/client/haproxy.py]:
```suggestion
        self.useProvidedWeights = Configuration().get("CLIENT_USE_WEIGHTS_PROVIDED_BY_HSDO") == "true"
```

**Review by tanguyfalconnet — CHANGES_REQUESTED** [review]

**Comment by vgallissot** [comment]
As we've discussed alltogether, we won't go further on this PR as we've decided that AgentCheck won't be very usefull in our case.

**Pull request closed without merging**